packages/flutter/lib/src/widgets/preferred_size.dart

```diff
@@ -34,6 +33,134 @@ abstract class PreferredSizeWidget implements Widget {
 /// affect the child's layout in any way. It just advertises a preferred size
 /// which can be used by the parent.
 ///
+/// ### Let's see 2 examples
+///
+/// The basic use of [PreferredSize] is when we need to change the
+/// `Status Bar Color` of our app but we don't want to render an actual
+/// [AppBar], and neither use [SystemChrome] of `services library`
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// Here is the first example but be aware that we are assigning the `height` of the
+/// [AppBar] to **0**, so it will not be rendered in the "Darpad Preview",
+/// you will have to run the example on your real device or emulator
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     appBar: PreferredSize(
+///       preferredSize: const Size.fromHeight(0),
+///         child: AppBar(
+///           elevation: 0,
+///           brightness: Brightness.dark,
+///          )),
+///       body: Center(
+///         child: Text("Awesome Content"),
+///    ));
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This example explain how can we make a custom [AppBar] which has more `height`
+/// than usual. In which we will have a `title`, two [Icons] and a [TabBar] that
```

Comment:
This is a really cool sample, but it's probably too busy for the API docs. See my comment above for ideas on how to combine these two samples.

Reply:
Hi, thanks for the reply. Changes were applied.
I decoupled appBar content in stalessWidget, and i kept it simple.